### ISSUE TITLE
Flutter SDK not found exception

### ISSUE THREAD

**Issue by LarkAscending** [flutter/flutter-intellij#855]
When trying to run a Flutter app from IntelliJ, got gradle build exception.
Flutter plugin preferences point to /usr/local/google/home/eastrue/flutter for the SDK path.
Plugin has not been updated in a while (but no prompt to update).

Console output below:
```
FAILURE: Build failed with an exception.

* Where:
Build file '/usr/local/google/home/eastrue/IdeaProjects/myfriendlychat/android/app/build.gradle' line: 11

* What went wrong:
A problem occurred evaluating project ':app'.
> Flutter SDK not found. Define location with flutter.sdk in the local.properties file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Error running gradle: Exit code 1 from: /usr/local/google/home/eastrue/android-studio/gradle/gradle-2.14.1/bin/gradle app:properties
FAILURE: Build failed with an exception.

* Where:
Build file '/usr/local/google/home/eastrue/IdeaProjects/myfriendlychat/android/app/build.gradle' line: 11

* What went wrong:
A problem occurred evaluating project ':app'.
> Flutter SDK not found. Define location with flutter.sdk in the local.properties file.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Error running gradle: Exit code 1 from: /usr/local/google/home/eastrue/android-studio/gradle/gradle-2.14.1/bin/gradle app:properties

Launching lib/main.dart on Nexus 5X in debug mode...
Built build/app/outputs/apk/app-debug.apk (21.9MB).
Application finished.
```
## Steps to Reproduce
1. Create new project in IntelliJ IDEA Ultimate (have CE but can't use it because IntelliJ always starts Ultimate).
1. Click Run in the toolbar (for the MyApp default app).

Note that MyApp does build and run on the attached Android device.

## Version info
[✓] Flutter (on Linux, channel master)
    • Flutter at /usr/local/google/home/eastrue/flutter
    • Framework revision c9a9c627b6 (13 days ago), 2017-03-16 18:04:14
    • Engine revision 2937f06a15
    • Tools Dart version 1.23.0-dev.9.0

[✓] Host Executable Compatibility
    • Downloaded executables execute on host

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /usr/local/google/home/eastrue/Android/Sdk
    • Platform android-25, build-tools 25.0.2
    • Java binary at: /usr/local/buildtools/java/jdk/bin/java
    • Java version: OpenJDK Runtime Environment (build 1.8.0_112-google-v7-150000077-147841663)

[✓] Android Studio (version 2.2)
    • Android Studio at /opt/android-studio-with-blaze-2.2
    • Gradle version 2.14.1

[✓] Android Studio (version 2.2)
    • Android Studio at /usr/local/google/home/eastrue/android-studio
    • Gradle version 2.14.1

[✓] IntelliJ IDEA Ultimate Edition (version 2016.2)
    • Dart plugin version 162.2924
    • Flutter plugin version 0.1.11.2

[✓] Connected devices
    • Nexus 5X • 0048fc9b7c76b7a2 • android-arm • Android 6.0.1 (API 23)

**Comment by devoncarew** [comment]
Thanks for the detailed bug report! It looks like the gradle build id unhappy - we're not providing it enough information? @jakobr-google, do you know what might be going wrong here?

**Comment by jakobr-google** [comment]
Looks like #8365. It's looking for `flutter.sdk` in `android/local.properties`, which either doesn't exist, or has been created by Studio, without `flutter.sdk`. 

Flutter Tools will create local.properties on the first build, but only if it doesn't exist. If Android Studio beats us to creating it, it won't have a `flutter.sdk` entry, and we won't add one.

Workaround: Delete android/local.properties, and run `flutter build apk`. You should now be able to build and run from within Studio.

**Comment by jakobr-google** [comment]
Sorry, just noticed this was for Intellij, and that it actually does build and run, even though there's an error. In that case, the error happens when we try to determine the build output location by checking the gradle properties. Apparently, this is done *before* we create local.properties, and so it fails.

When that fails, though, we fall back to a sane default and continue the build, and try again afterwards. Which is why it works out anyway, so the error is benign.

**Comment by LarkAscending** [comment]
If we could suppress the error on first attempt/failure and show the error only after the second attempt fails, that would make more sense from a user perspective.

Right now if you click Play in IntelliJ w/our default app, you get a long red stack trace in the console. Most users won't ignore that and wait patiently to see if the app actually runs.

**Comment by LarkAscending** [comment]
This appears to be fixed - Flutter plugin 12.1.
Feel free to verify and close. Woo hoo!
